src/main/java/com/unstoppabledomains/resolution/naming/service/CNS.java

```diff
@@ -84,6 +84,16 @@ public  String getOwner(String domain) throws NamingServiceException {
     return util.toList(rawData);
   }
 
+  @Override
+  public String getTokenUri(BigInteger tokenID) throws NamingServiceException {
+    try {
+      String tokenURI = proxyReaderContract.getTokenUri(tokenID);
+      return tokenURI;
+    } catch (Exception e) {
```

Comment:
@DancingAxolotl It worth checking whether this error means that domain is not registered. We could get an error just in case of connection issues. In this case we need to throw original error to the client.